### PR TITLE
Public facing reference repository for client side

### PR DESCRIPTION
**Pull request by hdogan-sf**
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000UQJaYAO/view
W-9773835

StripeClientSide directory and all related components were added here using existing components in sfdx-stripe repo.
Minor changes on setup.ts to make the following command work successfully:
sfdx commerce:payments:quickstart:setup -p StripeClientSide

This command will setup Stripe client side payment gateway so that our client can test it without dealing with unnecessary details.

### CODE REVIEW

File: src/commands/commerce/payments/quickstart/setup.ts [hunk 1]
```diff
@@ -43,19 +43,32 @@ export class PaymentsQuickstartSetup extends SfdxCommand {
...
         this.ux.log(
             msgs.getMessage('quickstart.setup.settingUpGatewayConvertingNamedCredentialsGatewayAdapterApex', [
                 selection,
             ])
         );
-        this.ux.log(JSON.stringify(shellJsonSfdx(`sfdx force:mdapi:convert -r ${examplesDir}`), null, 4));
+        const storeDir = STORE_DIR(undefined, 'ceo@mydevhub.com', 'demo@1commerce.com', '1commerce');
+
         this.ux.log(msgs.getMessage('quickstart.setup.pushingNamedCredentialsAndGatewayAdapterApexToOrg'));
-        this.ux.log(JSON.stringify(shellJsonSfdx(`sfdx force:source:push -f -u "${this.org.getUsername()}"`), null, 4));
```
Comment (tarcang) [inline src/commands/commerce/payments/quickstart/setup.ts:58]: Hello! Why did you change the convert-push to use deploy?

File: src/commands/commerce/payments/quickstart/setup.ts [hunk 2]
```diff
@@ -43,19 +43,32 @@ export class PaymentsQuickstartSetup extends SfdxCommand {
...
         ).toLowerCase()}/checkout/payment-gateway-integration/${selection[0].toUpperCase() + selection.substr(1)}/`;
+
         this.ux.log(
             msgs.getMessage('quickstart.setup.settingUpGatewayConvertingNamedCredentialsGatewayAdapterApex', [
                 selection,
             ])
         );
-        this.ux.log(JSON.stringify(shellJsonSfdx(`sfdx force:mdapi:convert -r ${examplesDir}`), null, 4));
+        const storeDir = STORE_DIR(undefined, 'ceo@mydevhub.com', 'demo@1commerce.com', '1commerce');
+
```
[the remark this answers is not in the document]
Comment (hdogan-sf) [reply src/commands/commerce/payments/quickstart/setup.ts:59]: I'll declare these hardcoded values as variables, thanks for the suggestion. 
This line is just to get the store directory under BASE_DIR (homedir + '/.commerce') based on the given parameters.

File: src/commands/commerce/payments/quickstart/setup.ts [hunk 3]
```diff
@@ -43,19 +43,32 @@ export class PaymentsQuickstartSetup extends SfdxCommand {
...
         const paymentGatewayName = `${selection}PG`;
+
         const examplesDir = `${EXAMPLE_DIR}/${StoreQuickstartSetup.getStoreType(
             this.org.getUsername(),
             this.flags['store-name'],
             this.ux
         ).toLowerCase()}/checkout/payment-gateway-integration/${selection[0].toUpperCase() + selection.substr(1)}/`;
+
         this.ux.log(
             msgs.getMessage('quickstart.setup.settingUpGatewayConvertingNamedCredentialsGatewayAdapterApex', [
```
[the remark this answers is not in the document]
Comment (hdogan-sf) [reply src/commands/commerce/payments/quickstart/setup.ts:54]: Good point Matt. I'll remove it.

**Review by tarcang — APPROVED** [review]

File: src/commands/commerce/payments/quickstart/setup.ts [hunk 4]
```diff
@@ -43,19 +43,17 @@ export class PaymentsQuickstartSetup extends SfdxCommand {
         const paymentGatewayAdapterName = `${selection}Adapter`;
         const paymentGatewayProviderName = `${selection}PGP`;
         const paymentGatewayName = `${selection}PG`;
+        const hubOrgAdminUsername = 'ceo@mydevhub.com';
```
Comment (mkosak-sf) [inline src/commands/commerce/payments/quickstart/setup.ts:46]: These are still hardcoded.  They should be sourced from the devhub config.  @tarcang Can you or someone from the team advis @hdogan-sf on how to do this?

Comment (mkosak-sf) [reply src/commands/commerce/payments/quickstart/setup.ts:46]: actually store name and scratch org username can potentially come as command input.

**Review by mkosak-sf — APPROVED** [review]

File: examples/b2c/checkout/payment-gateway-integration/StripeClientSide/classes/QueryUtils.cls [hunk 5]
```diff
@@ -0,0 +1,292 @@
+// =============================================================================================
```
Comment (tarcang) [inline examples/b2c/checkout/payment-gateway-integration/StripeClientSide/classes/QueryUtils.cls:1]: We had forceDataSoql method from sfdx utils class (you can look it for reference). Does this class do something particularly different?

**Review by tarcang — APPROVED** [review]

File: examples/b2c/checkout/payment-gateway-integration/StripeClientSide/classes/QueryUtils.cls (hunk 5 above)
Comment (mkosak-sf) [reply examples/b2c/checkout/payment-gateway-integration/StripeClientSide/classes/QueryUtils.cls:1]: This is an apex class used by the sample payment reference implementation that we added.  It has nothing to do with plugin/commands.  It gets deployed to the org when you run `sfdx commerce:payments:quickstart:setup --payment-adapter StripeClientSide`

**Review by unandyala — APPROVED** [review]

**Comment by hdogan-sf** [comment]
Closing this PR in favor of the following PR:
https://github.com/forcedotcom/commerce-on-lightning/pull/118/
